_posts/2020-05-15-guard-duty-orgs.md

```diff
@@ -0,0 +1,246 @@
+```
+
+Then we can explicitly set the region using:
+
+```typescript
+AWS.config.region = targetRegion
+```
+
+We're now ready to create the Guard Duty client for that region and account:
+
+```typescript
+const organisationsMasterGD = new AWS.GuardDuty()
+```
+
+Now that we have a Guard Duty client specific to that region and account, we can elect a Guard Duty master account with the `enableOrganizationAdminAccount` method:
+
+```typescript
+await organisationsMasterGD
+  .enableOrganizationAdminAccount({
+    AdminAccountId: params.GuardDutyAdminAccountId,
+  })
+  .promise()
+```
+
+As long as that call didn't throw an error, we can assume it was successful.
+
+## An aside on account strategies
+
+You might now be wondering why the Guard Duty master account would be a different account to the Organisations master account. To understand this we need to introduce the concept of account strategies...
+
+If you're working within an organisation with several external or internal projects and have people working in many different roles, it's likely you'll want to have some bulkheads between projects and workloads. This reduces the risk of one project accidentally clobbering or causing an outage for another project. It can also limit the blast radius of compromised systems.
+
+In AWS this is achieved by using AWS Organisations (and other supporting services) alongside an account strategy.
+
+An account strategy is a pre-determined approach for how you will structure and organise accounts. For example; `super-cool-crm-product` might run in two accounts; `super-cool-crm-product-non-prod` and `super-cool-crm-product-prod`, but `super-cool-website` might run in two other accounts; `super-cool-website-prod` and `super-cool-website-nonprod`.
+
+The examples above are what we call "workload" accounts i.e. they are running production or non-production systems depended on by customers or internal to the business. Depending on your needs, you may want to create other kinds of accounts, for example:
+
+* Security or Audit account for aggregating security findings
+* Log archive account for aggregating CloudTrail and CloudWatch logs
+* Build account for CI/CD pipelines
+
+Using AWS Organisations, these accounts can be organised into Organisational Units (OUs), which further allows us to apply policies to whole groups of accounts (e.g. ensuring CodePipeline is only used in the build accounts, and not workload accounts).
+
+In this instance, we are electing the Security/Audit account as the Guard Duty master, as it makes sense to aggregate security findings there.
+
+## Creating members
+
+Before we can create members, we need to get the ID of the newly created detector in the Guard Duty master account.
+
+Let's create a new Guard Duty client for the Guard Duty master account:
+
+```typescript
+AWS.config.credentials = new AWS.SharedIniFileCredentials({ profile: 'my-guard-duty-master-account-profile' })
```

Comment:
As a general rule I'd get out of the habit of setting this globally. If you are assuming roles across different accounts this can cause confusion. Normally I would do;

```
const credentials= new AWS.SharedIni...
const guardDuty = new GuardDuty({ credentials })
```

Reply:
Yeah that's a much better approach